nixos/modules/system/boot/networkd.nix

```diff
@@ -207,6 +282,142 @@ let
+        (assertValueOneOf "UDP6ZeroChecksumRx" boolValues)
+        (assertPort "DestinationPort")
+        (assertInt "FlowLabel")
+        (assertRange "FlowLabel" 0 1048575)
+        (assertValueOneOf "IPDoNotFragment" (boolValues + ["inherit"]))
+        (assertValueOneOf "Independent" boolValues)
+      ];
+
+      sectionBareUDP = checkUnitConfig "BareUDP" [
+        (assertOnlyFields [
+          "DestinationPort"
+          "EtherType"
+        ])
+        (assertHasField "DestinationPort")
+        (assertPort "DestinationPort")
+        (assertHasField "EtherType")
+        (assertValueOneOf "EtherType" ["ipv4" "ipv6" "mpls-uc" "mpls-mc"])
+      ];
+
+      sectionL2TP = checkUnitConfig "L2TP" [
+        (assertOnlyFields [
+          "TunnelId"
+          "PeerTunnelId"
+          "Remote"
+          "Local"
+          "EncapsulationType"
+          "UDPSourcePort"
+          "UDPDestinationPort"
+          "UDPChecksum"
+          "UDP6ZeroChecksumTx"
+          "UDP6ZeroChecksumRx"
+        ])
+        (assertHasField "TunnelId")
+        (assertRange "TunnelId" 1 4294967295)
+        (assertHasField "PeerTunnelId")
+        (assertRange "PeerTunnelId" 1 4294967295)
+        (assertHasField "Remote")
+        (assertHasField "Local")
+        (assertValueOneOf "EncapsulationType" ["udp" "ip"])
+        (assertPort "UDPSourcePort")
+        (assertPort "UDPDestinationPort")
+        (assertValueOneOf "UDPChecksum" boolValues)
+        (assertValueOneOf "UDP6ZeroChecksumTx" boolValues)
+        (assertValueOneOf "UDP6ZeroChecksumRx" boolValues)
+      ];
+
+      sectionL2TPSession = checkUnitConfig "L2TPSession" [
+        (assertOnlyFields [
+          "Name"
+          "SessionId"
+          "PeerSessionId"
+          "Layer2SpecificHeader"
+        ])
+        (assertHasField "Name")
+        (assertHasField "SessionId")
+        (assertRange "SessionId" 1 4294967295)
+        (assertHasField "PeerSessionId")
+        (assertRange "PeerSessionId" 1 4294967295)
+        (assertValueOneOf "Layer2SpecificHeader" ["none" "default"])
+      ];
+
+      sectionMACsec = checkUnitConfig "MACsec" [
+        (assertOnlyFields [
+          "Port"
+          "Encrypt"
+        ])
+        (assertPort "Port")
+        (assertValueOneOf "Encrypt" boolValues)
+      ];
+
+      sectionMACsecReceiveChannel = checkUnitConfig "MACsecReceiveChannel" [
+        (assertOnlyFields [
+          "Port"
+          "MACAddress"
+        ])
+        (assertHasField "Port")
+        (assertPort "Port")
+        (assertHasField "MACAddress")
+        (assertMacAddress "MACAddress")
+      ];
+
+      # NOTE The Key directive is missing on purpose here, please
+      # do not add it to this list. The nix store is world-readable let's
+      # refrain ourselves from providing a footgun.
+      sectionMACsecTransmitAssociation = checkUnitConfig "MACsecTransmitAssociation" [
+        (assertOnlyFields [
+          "PacketNumber"
```

Comment:
missing a range check (same for receive)

Reply:
I haven't seen a check that could work for `Takes a value between 1-4,294,967,295.`?

Reply:
`(assertRange "PacketNumber" 1 4294967295)` should work. nix ints are 64 bit, so there's no wrapping from signed/unsignedness here.